00-RELEASENOTES

```diff
@@ -1,16 +1,284 @@
+* Optimize `ZRANGE` offset location from linear search to skip list jump. (Redis#12450)
+* Fix `aeSetDontWait` timing to avoid unnecessary waits in `aeProcessEvent`. (Redis#12068)
+
+Performance/Efficiency Improvements - Cluster
+=============================================
+* Add lightweight cluster message header for Pub/Sub messages. (#654)
+* Minor performance improvement in Valkey cluster by avoid initializing key
+  buffer in `getKeysResult`. (#631)
+* Cache `CLUSTER SLOTS` response to improve throughput and reduce latency. (#53)
+* Replace slots_to_channels radix tree with slot-specific dictionaries for
+  shard channels. (Redis#12804)
+* Optimize `KEYS` command when pattern includes hashtag and implies a single
+  slot. (Redis#12754)
+* Optimize `SCAN` command with `MATCH` when pattern implies a single slot.
+  (Redis#12536)
+* Replace cluster metadata with slot specific dictionaries to reduce memory
+  usage when using Valkey cluster. (Redis#11695, Redis#12704)
+
+Reliability Improvements - Core
+===============================
+* Limit tracking custom errors (e.g. from Lua) while allowing normal errors
+  to be tracked (#500, Redis#13141)
+* Manage maximum number of new connections per cycle to prevent connection
+  storms. (Redis#12178)
+
+Reliability Improvements - Cluster
+==================================
+* Reduce fail-over time in Valkey cluster when multiple sequential fail-overs
+  occurred by resetting `failover_auth_time` when the new primary node goes
+  down. (#782)
+* Restrict node failure marking to primaries with assigned slots. (#634)
+* Enhance cluster meet reliability under link failures. (#461)
+* Improve reliability of slot migration in Valkey clusters. (#445)
+
+Usability Improvements - Core
+=============================
+* Re-brand and refine latency report messages. (#644)
+* Optimize `ACL LOAD` to avoid disconnecting clients whose users are unchanged.
+  (Redis#12171)
+
+Usability Improvements - Cluster
+================================
+* Adjust log levels for various cluster-related logs to improve clarity. (#633)
+* Maintain deterministic ordering of replica(s) in `CLUSTER SLOTS` response. (#265)
+
+Usability Improvements - CLI
+================================
+* Add prompt message when Ctrl-C is pressed in `valkey-cli`. (#702)
+* Keep an in-memory history of all commands in `valkey-cli` so that sensitive
+  commands can be shown within the same session. (Redis#12862)
+
+Module Improvements - Core
+==========================
+* Add `ValkeyModule_TryCalloc()` and `ValkeyModule_TryRealloc()` to handle
+  allocation failures gracefully. (Redis#12985)
+* Make `ValkeyModule_Yield` thread-safe by handling events in the main thread.
+  (Redis#12905)
+* Allow modules to declare new ACL categories. (Redis#12486)
+
+Module Improvements - Cluster
+=============================
+* Add API `ValkeyModule_ClusterKeySlot` and `ValkeyModule_ClusterCanonicalKeyNameInSlot`.
+  (Redis#13069)
+
+Behavior Changes - Core
```

Comment:
Another one here from Redis#12961. "Revert multi OOM limit and add multi buffer limit ", Not a good name. The gist of this one was that we made it so that Mulit execs would oom. Maybe some wording like: "Allow multi-exec to use a little bit of extra memory above the used-memory limit."